custom_components/mikrotik_router/mikrotik_controller.py

```diff
@@ -634,3 +665,229 @@ def get_queue(self):
+                {"name": "address", "default": "unknown"},
+                {"name": "host-name", "default": "unknown"},
+                {"name": "status", "default": "unknown"},
+                {"name": "last-seen", "default": "unknown"},
+                {"name": "server", "default": "unknown"},
+                {"name": "comment"},
+            ],
+            ensure_vals=[
+                {"name": "interface"},
+                {"name": "available", "type": "bool", "default": False},
+            ]
+        )
+
+        for uid in self.data["dhcp"]:
+            self.data["dhcp"][uid]['interface'] = \
+                self.data["dhcp-server"][self.data["dhcp"][uid]['server']]["interface"]
+
+            self.data["dhcp"][uid]['available'] = \
+                self.api.arp_ping(self.data["dhcp"][uid]['address'], self.data["dhcp"][uid]['interface'])
+
+    def _address_part_of_local_network(self, address):
+        address = ip_address(address)
+        for network in self.local_dhcp_networks:
+            if address in network:
+                return True
+        return False
+
+    def _get_accounting_mac_by_ip(self, requested_ip):
+        for mac, vals in self.data['accounting'].items():
+            if vals.get('address') is requested_ip:
+                return mac
+        return None
+
+    def get_accounting(self):
+        """Get Accounting data from Mikrotik"""
+        # Check if accounting and account-local-traffic is enabled
+        accounting_enabled, local_traffic_enabled = self.api.is_accounting_and_local_traffic_enabled()
+        traffic_type, traffic_div = self._get_traffic_type_and_div()
+
+        if not accounting_enabled:
+            # If any hosts were created return counters to 0 so sensors wont get stuck on last value
+            for mac, vals in self.data["accounting"].items():
+                self.data['accounting'][mac]["tx-rx-attr"] = traffic_type
+                if 'wan-tx' in vals:
+                    self.data["accounting"][mac]['wan-tx'] = 0.0
+                if 'wan-rx' in vals:
+                    self.data["accounting"][mac]['wan-rx'] = 0.0
+                if 'lan-tx' in vals:
+                    self.data["accounting"][mac]['lan-tx'] = 0.0
+                if 'lan-rx' in vals:
+                    self.data["accounting"][mac]['lan-rx'] = 0.0
+            return
+
+        # Build missing hosts from already retrieved DHCP Server leases
+        for mac, vals in self.data["dhcp"].items():
+            if mac not in self.data["accounting"]:
+                self.data["accounting"][mac] = {
+                    'address': vals['address'],
+                    'mac-address': vals['mac-address'],
+                    'host-name': vals['host-name'],
+                    'comment': vals['comment']
+                }
+
+        # Build missing hosts from already retrieved ARP list
+        host_update_from_arp = False
+        for entry in self.raw_arp_entries:
```

Comment:
lets create and use new api_parse in new async function get_arp for arp self.data["arp"] so it can be reused elsewhere.
you can rename current self.data["arp"] I use in get_interface_client to self.data["arp_temp"]. I will redo it later, since its still done in old way and we should have consistency.

Reply:
available as self.data["arp"]
old arp renamed to arp_temp, I will remove it later